ntbs-service/Pages/Notifications/Edit/Comorbidities.cshtml.cs

```diff
@@ -18,28 +24,62 @@ public class ComorbiditiesModel : NotificationEditModelBase
         protected override async Task<IActionResult> PrepareAndDisplayPageAsync(bool isBeingSubmitted)
         {
             ComorbidityDetails = Notification.ComorbidityDetails;
+            ImmunosuppressionDetails = Notification.ImmunosuppressionDetails;
 
             await SetNotificationProperties(isBeingSubmitted, ComorbidityDetails);
+            await SetNotificationProperties(isBeingSubmitted, ImmunosuppressionDetails);
+
             if (ComorbidityDetails.ShouldValidateFull)
             {
-                TryValidateModel(ComorbidityDetails, ComorbidityDetails.GetType().Name);
+                TryValidateModel(ComorbidityDetails, nameof(ComorbidityDetails));
+            }
+
+            if (ImmunosuppressionDetails.ShouldValidateFull)

```

Comment:
Minor comment - could probably move these to the same if loop (whether you still check off both or just one should be irrelevant AFAIK)